tap_google_sheets/transform.py

```diff
@@ -145,37 +145,77 @@ def transform_sheet_boolean_data(value, unformatted_value, sheet_title, col_name
+    numeric_value = formatted_value.replace(",", "")
+    try:
+        # Verify we can convert formatted value to float for scientific formatted numbers
+        # For example:
+        #   formatted value: "1.23E+03"
+        #   unformatted value: 1234
+        # thus, we can convert "1.23E+03" to float but, for int casting we get error and wrong value will be returned
+        float(numeric_value)
+    except ValueError:
+        return str(formatted_value) # Return original value in case of ValueError
+
+    if value_type == int:
+        return int(unformatted_value)
+
+    # For float type data, round off to 15 decimal digits
     # Determine float decimal digits
-    decimal_digits = str(value)[::-1].find('.')
+    decimal_digits = str(unformatted_value)[::-1].find('.')
     if decimal_digits > 15:
         try:
             # ROUND to multipleOf: 1e-15
-            col_val = float(round(value, 15))
+            col_val = float(round(unformatted_value, 15))
         except ValueError:
-            col_val = str(value)
+            col_val = str(unformatted_value)
             LOGGER.info('WARNING: POSSIBLE DATA TYPE ERROR; SHEET: {}, COL: {}, CELL: {}{}, TYPE: {}'.format(
                 sheet_title, col_name, col_letter, row_num, col_type))
         return col_val
     else: # decimal_digits <= 15, no rounding
         try:
-            col_val = float(value)
+            col_val = float(unformatted_value)
         except ValueError:
-            col_val = str(value)
+            col_val = str(unformatted_value)
             LOGGER.info('WARNING: POSSIBLE DATA TYPE ERROR: SHEET: {}, COL: {}, CELL: {}{}, TYPE: {}'.format(
                 sheet_title, col_name, col_letter, row_num, col_type))
         return col_val
 
 # transform number values in the sheet
-def transform_sheet_number_data(value, sheet_title, col_name, col_letter, row_num, col_type):
-    if type(value) == int:
-        return int(value)
-    elif type(value) == float:
-        return transform_sheet_decimal_data(value, sheet_title, col_name, col_letter, row_num, col_type)
+def transform_sheet_number_data(formatted_value, unformatted_value, sheet_title, col_name, col_letter, row_num, col_type):
+    """
+        Handled number type records by type-casting into a float to verify the user has entered number type data.
+
+        :param
+        formatted_value - The displayed value of a cell in the sheet
+        unformatted_value - The formatted value of a cell in the sheet
+        sheet_title - The title of the sheet
+        col_name - Column name
+        col_letter - Column letter of the record ie. A, B, C, etc.
+        row_num - Row number of the record
+        col_type - Column type of the record (here: numberType)
+    """
+    if type(unformatted_value) in [int, float]:
+        return transform_sheet_decimal_data(type(unformatted_value), formatted_value, unformatted_value, sheet_title, col_name, col_letter, row_num, col_type)
```

Comment:
@harshpatel4crest We should try to keep less number of function arguments for a cleaner code.
We can get the type of unformatted value in the calling function as well. Can you please remove it and define a local variable in `transform_decimal_data` evaluating the type of unformatted value?

Reply:
Updated the code to use **type(unformatted_value)** rather than explicitly passing as an argument. The arguments: `sheet_title, col_name, col_letter, row_num, col_type` are used for logging purpose only.